File: cs12b/pa3/DictionaryTest.java
/* Author: Daniel Boles
 * Project: PA3
 * Class: 12b
 */

public class DictionaryTest{

   public static void main(String[] args){
      String v;
      Dictionary A = new Dictionary();

      System.out.println(A.isEmpty());
      System.out.println(A.size());
      A.insert("1", "1");
      A.insert("2", "Q");
      System.out.println(A);

      v = A.lookup("1");
      System.out.println("key=1 "+(v==null?"not found":("value="+v)));
      v = A.lookup("3");
      System.out.println("key=3 "+(v==null?"not found":("value="+v)));

      A.insert("2","Q");  // causes DuplicateKeyException

      System.out.println(A.isEmpty());
      System.out.println(A.size());
      A.delete("1");
      System.out.println(A);

      A.delete("5");  // causes KeyNotFoundException

      A.makeEmpty();
      System.out.println(A.isEmpty());
      System.out.println(A);

   }
}

File: cs12b/lab5/DictionaryTest.c
//-----------------------------------------------------------------------------
// DictionaryTest.c
// Test client for the Dictionary ADT
// Author: Daniel Boles
//-----------------------------------------------------------------------------

#include<stdio.h>
#include<stdlib.h>
#include<string.h>
#include"Dictionary.h"

#define MAX_LEN 180

int main(int argc, char* argv[]){
   Dictionary A = newDictionary();
   char* k;
   char* v;


   char* word1[] = {"1","2","3"};
   char* word2[] = {"cat","foobar","garlicbread"};
   int i;

   for(i=0; i<strlen(word1); i++){
      insert(A, word1[i], word2[i]);
   }

   printDictionary(stdout, A);
   makeEmpty(A);
   printDictionary(stdout, A);

   for(i=0; i<strlen(word1); i++){
         insert(A, word1[i], word2[i]);
      }

   for(i=0; i<strlen(word1); i++){
      k = word1[i];
      v = lookup(A, k);
      printf("key=\"%s\" %s\"%s\"\n", k, (v==NULL?"not found ":"value="), v);
   }

   insert(A, "4", "rat");

   delete(A, "2");

   printDictionary(stdout, A);

   freeDictionary(&A);

   return(EXIT_SUCCESS);
}

